presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/AddIntermediateAggregations.java

```diff
@@ -174,12 +175,15 @@ private PlanNode addGatheringIntermediate(AggregationNode aggregation, PlanNodeI
         for (Map.Entry<Symbol, AggregationNode.Aggregation> entry : assignments.entrySet()) {
             Symbol output = entry.getKey();
             AggregationNode.Aggregation aggregation = entry.getValue();
+            checkState(aggregation.getOrderBy().isEmpty() && aggregation.getOrdering().isEmpty(), "Intermediate aggregation does not support ORDER BY");
```

Comment:
Yes, I'm not sure whether there's guide line around this. Basically when two things are supposed to be always consistent, do we check for both or just one. I can see benefit in both. Either one is fine with me.